### PR TITLE
Welcome page

### LINKED ISSUE #15: Welcome Page
When a user visits the root path they should be on the welcome page which includes:

- [x] Welcome message
- [x] Brief description of the application
- [x] Button to Log in
- [x] Link to Registration

**Details:** Implement basic auth in the application allowing a user to log in with an email and password. The password should be stored in the database using [bcrypt](https://github.com/codahale/bcrypt-ruby).

### PR DESCRIPTION
**Pull request by araspaas**
Add Travis CI and initial gem setup
- Ran bundle install and bundle update.
- Test for Travis ci

Add feature welcome page
- Update test for welcome message to have more substance to it
- Update the welcome page with more information about the website
- Add tests for having a log in form on the welcome page as well as a link to register a new account
- Add the form to log in as well as a link to create a new account on the welcome page

Add feature to log in and log out
- Add user table migration
- Add bcrypt, shoulda-matchers gems and format for bcrypt
- Add user model spec for validations
- Add user model with validations
- Add sad path test for failing to log in, add a test for a user to login, add test that a visitor cant visit a dashboard
- Add sessions controller with new, create, destroy actions and login params method
- Add user to seed file 
- Add nav_spec file and test to log out as a user
- Add routes to log in/log out. Add dashboard index routes 
- Add flash message setup and a nav bar with logout link 
- Add dashboard index page 
- Add dashboard controller with index action and authorization 
- Add current user method to application_controller

closes #15

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+class SessionsController < ApplicationController
+
+  def new
+    flash[:notice] = "You are already logged in!" if current_user
```
Comment (smcunning) [inline app/controllers/sessions_controller.rb:4]: Let's remember to test for this once we start developing the dashboard page!

File: app/views/welcome/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<%= content_tag :h1, "Welcome To Viewing Party!" %>
+<%= content_tag :h3, "This website allows you to create a 'viewing party' with your friends to watch movies together online from the comfort of your own homes. Especially during these hard time during a pandemic when it isn't safe to be in the same room." %>
+
+
+<%= form_tag "/login" do %>
```
Comment (smcunning) [inline app/views/welcome/index.html.erb:5]: Like we discussed, possibly a refactor opportunity to use form_for or form_with instead of form_tag here.

**Review by smcunning — APPROVED** [review]
LGTM!

File: app/views/welcome/index.html.erb (hunk 2 above)
Comment (KieraAllen) [reply app/views/welcome/index.html.erb:5]: or form_with...? Aren't they replacing form_tag with that?

**Review by KieraAllen — APPROVED** [review]

**Pull request merged**